File: php-pr/51/php/code-quality-performance/SumCalculator.php
<?php

class SumCalculator
{
    public function buildMessage($words)
    {
        // {fact rule=code-quality-performance@v1.0 defects=1}
        $result = '';
        
        for ($i = 0; $i < count($words); $i++) {
            $result .= $words[$i] . ' ';
        }
        
        return trim($result);
        // {/fact}
    }
}

File: kotlin-pr/50/kotlin/code-quality-performance/SumCalculator.kt
class SumCalculator {
    fun buildMessage(words: List<String>): String {
        // {fact rule=code-quality-performance@v1.0 defects=1}
        var result = ""
        
        for (i in words.indices) {
            result += words[i] + " "
        }
        
        return result.trim()
        // {/fact}
    }
}